### PR TITLE
rbd-mirror: track images via global image id

### PR DESCRIPTION
**Pull request by dillaman**

### CODE REVIEW

**Comment by trociny** [comment]
@dillaman Don't you think we can observe the false alarms you fixed in  "qa/workunits/rbd" commits when testing our stable branches? I mean that it might make sense to backport these and a separate PR would make it easier?

Also, the commit log message "handle teuthology-specific race rbd-mirror race"  looks too racy.

**Comment by dillaman** [comment]
@trociny 

> Also, the commit log message "handle teuthology-specific race rbd-mirror race" looks too racy.

lol -- indeed.  I'll put both of those fixes into a backport PR.

**Comment by dillaman** [comment]
@trociny updates pushed

**Comment by dillaman** [comment]
@trociny The ```ImageReplayer``` class instances for the down images are complaining that "start: already running" every minute so I don't think it's related to any changes in this PR. I'll see if I can locally reproduce to figure out how it got stuck in a pseudo-started state for the primary images. Most likely some odd ImageReplayer shut down (due to the image being primary) race.

**Comment by dillaman** [comment]
@trociny In that case -- I'd hold off on merging this until I can repeat and ensure this isn't a regression.

**Comment by dillaman** [comment]
@trociny I've had the stress test running in a constant loop on my machine all day w/o fail until just now where the rbd CLI hung attempting to acquire the image status. It looks like it was a low-level objecter issue since it had a hung request that required restarting the OSD to recover. If the rbd-mirror daemon experienced a similar osdc / osd bug like I hit, that could possibly explain the error messages about "already started" if it was attempting to shut-down but librados was effectively non-responsive.

TL;DR: I think this PR is functional and I need to see if I can track down the osd / osdc issue that caused a permanently hung request.

**Comment by trociny** [comment]
@dillaman Was that hung permanent in your case? Because in that teuthology test failure it looked like rather a slowdown, not a permanent hung:
```
log$ zcat *mirr*gz |grep '619e3619-260d-48be-9712-3785f0c49e44.*start: already running' |sort
2017-02-14 22:56:11.928297 7f2465865700 -1 rbd::mirror::ImageReplayer: 0x7f23c80024f0 [2/619e3619-260d-48be-9712-3785f0c49e44] start: already running
2017-02-14 23:00:41.990556 7f2465865700 -1 rbd::mirror::ImageReplayer: 0x7f23c80024f0 [2/619e3619-260d-48be-9712-3785f0c49e44] start: already running
2017-02-14 23:01:11.997393 7f2465865700 -1 rbd::mirror::ImageReplayer: 0x7f23c80024f0 [2/619e3619-260d-48be-9712-3785f0c49e44] start: already running
2017-02-14 23:01:42.003361 7f2465865700 -1 rbd::mirror::ImageReplayer: 0x7f23c80024f0 [2/619e3619-260d-48be-9712-3785f0c49e44] start: already running
2017-02-14 23:03:12.018426 7f2465865700 -1 rbd::mirror::ImageReplayer: 0x7f23c80024f0 [2/619e3619-260d-48be-9712-3785f0c49e44] start: already running
2017-02-14 23:03:42.023971 7f2465865700 -1 rbd::mirror::ImageReplayer: 0x7f23c80024f0 [2/619e3619-260d-48be-9712-3785f0c49e44] start: already running
2017-02-14 23:04:12.029526 7f2465865700 -1 rbd::mirror::ImageReplayer: 0x7f23c80024f0 [2/619e3619-260d-48be-9712-3785f0c49e44] start: already running
2017-02-14 23:04:42.033766 7f2465865700 -1 rbd::mirror::ImageReplayer: 0x7f23c80024f0 [2/619e3619-260d-48be-9712-3785f0c49e44] start: already running
2017-02-14 23:15:12.340505 7f2465865700 -1 rbd::mirror::ImageReplayer: 0x7f23c80024f0 [2/619e3619-260d-48be-9712-3785f0c49e44] start: already running
```
Note, the errors were not observed permanently. For me it looked like there were periods of time when the image replayer was too slow in its periodic run (start -> check the image -> detect it is primary -> stop), which lasted for up to 1.5 minutes and during these periods 'already running' errors observed from the next (30 sec interval) attempts. I thought the cause of this slowdown might have been a mess introduced by the injected socket failures.
